tempfile.go

```diff
@@ -167,6 +168,16 @@ func (t *PendingFile) CloseAtomicallyReplace() error {
 		return err
 	}
 	t.done = true
+	if t.syncDirectory {
+		dh, err := os.OpenFile(filepath.Dir(t.path), os.O_WRONLY, 0)
```

Comment:
On Linux this doesn't work. (Error: `open <file>: is a directory`)

Simply `os.Open(filepath.Dir(t.path))` does work instead.

See also:
- https://linux.die.net/man/2/fsync
  > On some UNIX systems (**but not Linux**), fd must be a writable file descriptor.

- https://austin-group-l.opengroup.narkive.com/vC4Fjvsn/fsync-ing-a-directory-file-descriptor
  > On Linux, you would
typically open in read-only (O_RDONLY) the directory (O_WRONLY would
fail otherwise), and call fsync() on the returned file descriptor.

Reply:
This is a good point — could you add a test so that the feature is exercised in the CI pipeline?